src/Features/CSharp/Portable/CodeRefactorings/InvertIf/InvertIfCodeRefactoringProvider.cs

```diff
@@ -63,32 +64,125 @@ internal partial class InvertIfCodeRefactoringProvider : CodeRefactoringProvider
                 return;
             }
 
+            var invertedIfBody = SyntaxKind.None;
+            if (ifStatement.Else == null)
+            {
+                var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+                if (!CanInvertElselessIf())
+                {
+                    return;
+                }
+
+                bool CanInvertElselessIf()
+                {
+                    var controlFlow = semanticModel.AnalyzeControlFlow(ifStatement.Statement);
+                    return !controlFlow.EndPointIsReachable || (invertedIfBody = GetInvertedIfGeneretedBody()) != default;
```

Comment:
check against SyntaxKind.None.  This code makes it look like you're actually generating syntax.